tests/ip_api_test.go

```diff
@@ -0,0 +1,200 @@
+package tests
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	"github.com/phoenixnap/go-sdk-bmc/ipapi"
+	"github.com/stretchr/testify/suite"
+)
+
+type IpApiTestSuite struct {
+	suite.Suite
+	Client *ipapi.APIClient
+	Ctx    context.Context
+}
+
+func (suite *IpApiTestSuite) SetupSuite() {
```

Comment:
Shall we do the setup before every tests therefore having this behave like a before each? If so it has to be called SetupTest as per [unit test article for golang](https://medium.com/nerd-for-tech/writing-unit-tests-in-golang-part3-test-suite-6cca903be9ab). Imo it should behave as such cause we ideally reset the context every time we run a new test. Even though the teardown does some of the work with resetting expectations as well.

Reply:
im not really sure what logic you'd want to move into a before each? we're setting stuff that should be set once and left alone

Reply:
fair enough